src/main/java/com/marek/onlinebookstore/repository/book/BookRepository.java

```diff
@@ -12,12 +10,7 @@
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+
     @Query("SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id = :categoriesId")
     List<Book> findAllByCategoriesId(Long categoriesId, Pageable pageable);
-
-    @Query("SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id IS NOT NULL")
-    Page<Book> findAllWithCategories(Pageable pageable);
```

Comment:
The method findAllWithCategories has been deleted. If this method is not needed, make sure it is not being referenced anywhere else in the code to avoid potential errors.